src/cart/crt_iv.c

```diff
@@ -410,20 +410,21 @@ crt_ivf_finalize(struct iv_fetch_cb_info *iv_info, crt_iv_key_t *iv_key,
 
 			output = crt_reply_get(rpc);
 			output->ifo_rc = output_rc;
-
+			output->ifo_misc = *output_value;
 			rc = crt_reply_send(rpc);
+			memset(&output->ifo_misc, 0, sizeof(output->ifo_misc));
 			D_ASSERT(rc == 0);
 		}
 
 		/* addref done in crt_hdlr_iv_fetch */
 		RPC_PUB_DECREF(rpc);
 	} else {
 		iv_info->ifc_comp_cb(iv_info->ifc_ivns_internal,
-					iv_info->ifc_class_id,
-					iv_key, NULL,
-					iv_value,
-					output_rc,
-					iv_info->ifc_comp_cb_arg);
+				     iv_info->ifc_class_id,
+				     iv_key, NULL,
+				     output_rc == 0 ? iv_value : output_value,
```

Comment:
i dont think we should overload meaning of iv_value here. instead i was thinking ifc_comp_cb() should be expanded and along with output_rc, it would contain optional output_value that user can interpreset as they wish

Reply:
hmm, ok. let me try.